.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,142 @@
+---
+name: E2E Tests
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    paths-ignore:
+      - .circlecli/**
+      - meta/**
+      - .gitignore
+      - .markdownlint.yml
+      - .yamllint
+      - ansible.cfg.galaxy
+      - CHANGELOG.md
+      - LICENSE
+      - poetry.lock
+      - pyproject.toml
+      - README.md
+      - Vagrantfile
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: "0 1 * * *"
+  workflow_dispatch:
+
+env:
+  PY_COLORS: true
+  ANSIBLE_FORCE_COLOR: true
+  MOLECULE_DEBUG: false
+  MOLECULE_VERBOSITY: 0
+  MOLECULE_NO_LOG: true
+
+concurrency:
+  group: e2e
+  cancel-in-progress: false
+
+jobs:
+  python36:
+    runs-on: ubuntu-20.04
+    strategy:
+      max-parallel: 1
+      fail-fast: false
+      matrix:
+        python:
+          - version: 3.6
+            ansible-core: 2.11.12
+            molecule: 3.6.1
+            molecule-docker: 1.1.0
+        scenario:
+          - name: python36
+        st2_repo:
+          - name: stable
+          - name: unstable
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python.version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python.version }}
+
+      - name: Install Python dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade ansible-core==${{ matrix.python.ansible-core }}
+          python3 -m pip install --upgrade molecule==${{ matrix.python.molecule }}
+          python3 -m pip install --upgrade molecule-docker==${{ matrix.python.molecule-docker }}
+          ansible-galaxy collection install --requirements-file requirements.yml
+
+      - name: Molecule test ${{ matrix.st2_repo.name }} under ${{ matrix.scenario.name }}
+        run: |
+          molecule test --scenario-name ${{ matrix.scenario.name }}
+        env:
+          HUBOT_SLACK_TOKEN: ${{ secrets.HUBOT_SLACK_TOKEN }}
+          ST2_REPO: ${{ matrix.st2_repo.name }}
+
+  python38:
```

Comment:
Interesting split matrix between python3.6 vs python3.8.

Thinking that once we add, change or remove the python version we'll just have more places to change in this repo which is undesired as we want to minimize the support burden.
From another side, highlighting py3.6 vs py3.8 could be useful from the quick-look standpoint.

I think on the Ansible side we care more about the OS in the first place (as before) and it's probably good to leave the python versioning matrix to the st2 repo.

Reply:
I'm not sure that I agree here, if we support 3.6 and 3.8, we should test them both.

Reply:
We test python 3.6 and 3.8 in the https://github.com/stackStorm/st2 repository.

In the Ansible-st2 we care about the OS matrix (stable-el7, unstable-u20, etc).

Reply:
> We test python 3.6 and 3.8 in the https://github.com/stackStorm/st2 repository.

You're not testing *Ansible* with Python 3.6 and 3.8 in that repository, are you?

Reply:
What I'm getting at is that the purpose of this repository is to ensure that *Ansible* can deploy StackStorm using Python 3.6 and Python 3.8, which is why, again, I think that we should *test* that.

Reply:
Let's re-iterate. I'd like to understand what we're trying to test, why, and the intent.

Based on the latest message, are you trying to test that Ansible could be ran in an environment that has python3.6 or python3.8 versions installed?
That host with Ansible can potentially setup remote StackStorm instance based on the roles we have here.

Reply:
> what we're trying to test, why, and the intent

What this PR is attempting to verify is that the Ansible roles and playbooks in this repository work in deploying StackStorm, using Ansible (with both Python 3.6 and 3.8 *locally*), to CentOS 7, Rocky Linux 8, Ubuntu 18, and Ubuntu 20. I think where we're crossing wires is that what I'm introducing here is validating the local combinations of Python 3.6 and 3.8 and their respective versions of Ansible Core. Again, what I think is important in regards to this repository is that it successfully uses Ansible and the playbooks and roles in this repository *locally* to deploy StackStorm *against* the supported operating systems. Since StackStorm supports Python 3.6 and 3.8 currently, I think it's important to test both of them *locally* in addition to *remotely*.

Reply:
In addition to the next big step of turning this repository into an Ansible collection, what we should do is be more formal in regards to the [requirements](https://github.com/StackStorm/ansible-st2#requirements). Having a matrix with *local* Python and Ansible combinations (as this PR introduces) will facilitate this, i.e., when a new version of Ansible Core is released, or when an older version of Ansible Core is no longer supported, it's simply a matter of bumping a couple of versions and all of a sudden we're testing on the new, and no longer testing on the old.

Reply:
> > > Yeah, that's why I have it running serially, basically. I'll think on how to mitigate the rate-limiting even more.
> > 
> > 
> > Please switch to the previous `{repo}-{os}` build matrix, there are several benefits behind that and it'll also help us with the rate-limiting naturally.
> 
> Sure, do you care about testing both Python 3.6 and 3.8 _locally_ (in CI) or do you want to just test 3.8?

[Here's](https://github.com/StackStorm/ansible-st2/pull/329) a draft PR which ignores 3.6 locally and matrixes across repo and OS, do you like this better?

Reply:
At a minimum you would expect our playbooks to be run on our target O/Ses to install StackStorm via ansible locally. So that users don't have to have a separate box to install on. Previously we were using random pythons in the ansible E2E as it was picking up various defaults, such as python3.9, python2.7 etc...

In that regard, then I think we'd need to be able to run with more than just python3.6 or python3.8 locally, as taking focal and bionic then they will have different python available.

Reply:
Per https://github.com/StackStorm/ansible-st2/pull/328#issuecomment-1369132002, Ansible client has a very wide py version support between `py2.7`-`py3.11`.

I think we're already in a good spot and can trust the py version compatibility advertised and tested by the Ansible project.